### PR TITLE
Refactor/make send tokens internal call

### PR DESCRIPTION
**Pull request by gcontarini**
Change sendTokens to be an internal method on both xVault and chainVault.

The already implemented tests on xVault will be broken.

### CODE REVIEW

File: src/vaults/SectorCrossVault.sol [hunk 1]
```diff
@@ -46,54 +42,57 @@ contract SectorCrossVault is SectorBase {
...
-
-				emit BridgeAsset(chainId, vault.chainId, amount);
-			}
+			totalChildHoldings += amount;
+			beforeWithdraw(amount, 0);
+
+			_sendMessage(
+				vaultAddr,
+				vault,
+				Message(amount, address(this), address(0), chainId),
```
Comment (0x0scion) [inline src/vaults/SectorCrossVault.sol:59]: TODO: pass the fee amount in as a param, validate and subtract from amount in message

**Review by 0x0scion — APPROVED** [review]

File: src/vaults/SectorCrossVault.sol (hunk 1 above)
Comment (gcontarini) [reply src/vaults/SectorCrossVault.sol]: What validate means in this context?

File: src/vaults/SectorCrossVault.sol [hunk 2]
```diff
@@ -46,54 +42,63 @@ contract SectorCrossVault is SectorBase {
...
+			totalAmount += amount;
+
+			_sendMessage(
+				vaultAddr,
+				vault,
+				Message(amount - fee, address(this), address(0), chainId),
+				messageType.DEPOSIT
+			);
+
+			// This is fucked but dont know why
```
Comment (0x0scion) [inline src/vaults/SectorCrossVault.sol:65]: rm comment

Comment (gcontarini) [reply src/vaults/SectorCrossVault.sol]: done

**Pull request merged**